crates/flox/src/commands/package.rs

```diff
@@ -369,10 +374,58 @@ impl Publish {
         info!("done!");
 
         info!("Checking binary can be downloaded from {substituter_url}...");
-        publish
-            .check_substituter(substituter_url)
-            .await
-            .context("Binary cannot be downloaded")?;
+
+        let mut retry_count = None;
+        let found_upstream = loop {
+            let check_result = publish.check_substituter(substituter_url.clone()).await;
+
+            if check_result.is_ok() {
+                break true;
+            }
+
+            let err = check_result.unwrap_err();
+            warn!("Unable to find binary at {substituter_url}: {err}");
+
+            // in first round:
+            // if we can prompt, confirm that we should retry
+            // else              assume yes, retry {self.retry} times
+            if retry_count.is_none() && Dialog::can_prompt() {
+                let help_message = Some(format!(
+                    "flox query {substituter_url} {} more times; waiting 5 seconds inbetween.",
```

Comment:
I'm not sure what flox query means?
```suggestion
                    "query {substituter_url} {} more times, waiting 5 seconds in-between.",
```

Reply:
should be "quer**ies**" or "probes"?

Reply:
I think there's two ways to say it:
- "flox _will_ query <URL> ..."
- "query <URL> ..."